internal/privi/server.go

```diff
@@ -425,4 +426,102 @@ func (s *Server) ListNotifications(w http.ResponseWriter, r *http.Request) {
 		utils.LogAndHTTPError(w, err, "encoding response", http.StatusInternalServerError)
 		return
 	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *Server) CreateNotification(w http.ResponseWriter, r *http.Request) {
+	_, ok := s.getAuthedUser(w, r)
+	if !ok {
+		return
+	}
+	input := habitat.NetworkHabitatNotificationCreateNotificationInput{}
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		utils.LogAndHTTPError(w, err, "reading request body", http.StatusInternalServerError)
+		return
+	}
+	log.Info().Msgf("body: %s", string(body))
+	err = json.Unmarshal(body, &input)
+	if err != nil {
+		utils.LogAndHTTPError(w, err, "decode json request", http.StatusBadRequest)
+		return
+	}
+
+	log.Info().Msgf("input: %+v", input)
+
+	body, err = json.Marshal(input)
+	if err != nil {
+		utils.LogAndHTTPError(w, err, "marshal json request", http.StatusInternalServerError)
+		return
+	}
+
+	did, dpopClient, ok := s.oauthServer.Validate(w, r)
+	if !ok {
+		return
+	}
+	// Try handling both handles and dids
+	atid, err := syntax.ParseAtIdentifier(did)
+	if err != nil {
+		utils.LogAndHTTPError(w, err, "failed to parse at identifier", http.StatusBadRequest)
+		return
+	}
+	id, err := s.dir.Lookup(r.Context(), *atid)
+	if err != nil {
+		utils.LogAndHTTPError(w, err, "failed to lookup identity", http.StatusBadRequest)
+		return
+	}
+	pdsUrl, ok := id.Services["atproto_pds"]
```

Comment:
if the dpop client doesn't already know the pds location, then it should. it's only used for talking to the pds so it should automatically route there